src/utils/configfile.cc

```diff
@@ -81,14 +81,22 @@ std::string ConfigFile::ProcessConfigFlag(
 bool ConfigFile::Init(OptionsParser* options) {
   arguments_.clear();
 
-  // Get the relative path from the config file parameter.
+  // Get the path from the config file parameter.
   std::string filename = ProcessConfigFlag(CommandLine::Arguments());
 
-  // If filename is an empty string then return true.  This is to override
+  // If filename is an empty string then return true. This is to override
   // loading the default configuration file.
   if (filename == "") return true;
 
-  filename = CommandLine::BinaryDirectory() + "/" + filename;
+  // Check to see if we are using the default config file or not.
+  const bool using_default_config = 
+      filename == std::string(kDefaultConfigFile);
```

Comment:
I think it's better to have a separate value for default behaviour, like "`<default>`" or "`<auto>`" (which still will read `$LC0_DIRECTORY/lc0.config`), similarly to what we have for weights file (there we have "`<autodiscover>`").

Otherwise when run like `./lc0 -c lc0.cfg` it will search in the current directory, while for `./lc0 -c lc0.config` it will search in the directory of the binary, which is a confusing inconsistency.

Also in future we'll try several locations by default (possibly with different file names, not only directories, e.g. `/etc/lc0.cfg` vs `~/.lc0rc`), and it's more logical if the parameter doesn't have file name in it.

Reply:
I have made changes along these lines. Let me know if this is what you were thinking :slightly_smiling_face: